src/kudu/consensus/consensus_peers.cc

```diff
@@ -527,6 +526,9 @@ void Peer::ProcessResponseError(const Status& status) {
     return;
   }
 
+  // Increment failed attempts only when this is not an expected rejection by a
```

Comment:
rotation error goes through early return on lines 523-527 thus not incrementing failed_attempts_